elasticsearch/src/main/java/com/tcdi/zombodb/query_parser/QueryRewriter.java

```diff
@@ -379,33 +380,76 @@ private AggregationBuilder build(ASTTally agg) {
         }
     }
 
+    /**
+     * Determine if a particular field name is present in the index
+     *
+     * @param md index metadata
+     * @param fieldname field name to check for
+     * @return true if this field exists, false otherwise
+     */
+    private boolean hasDate(final IndexMetadata md, final String fieldname) {
+        return md.hasField(fieldname + DateSuffix);
+    }
+
     static class RangeSpecEntry {
         public String key;
         public Double from;
         public Double to;
     }
 
+    /**
+     * Container for date range aggregation spec
+     */
+    static class DateRangeSpecEntry {
```

Comment:
Ugh, upon further inspection, `AbstractRangeBuilder` doesn't declare the various `addRange()` methods, the concrete subclasses do, and they do it with different (and incompatible) signatures.

Maybe it'll just have to stay the way it is.  :/